CONTRIBUTING.md

```diff
@@ -21,6 +21,17 @@ please ensure that:
 2. The documentation has been updated accordingly. In particular, if a
    function or class has been modified during the PR, please update the
    *docstring* accordingly.
+
+   If your pull request is adding a new class, function, or module that is
+   intended to be user facing ensure that you've also added those to a
+   documentation `autosummary` index to include it in the api documentation.
+   For more details you can refer to:
+
+   https://qiskit.org/documentation/contributing_to_qiskit.html#documentation-structure
+
+   which documents how the documentation is structured and how you can add
```

Comment:
I think the last line of this sentence may be a bit redundant imho. I think it's fine to just say 'for more details you can refer to <link>.'
This reduces the cognitive overload a bit, meaning people may be less likely to skip over.